aspnetcore/blazor/forms-and-input-components.md

```diff
@@ -37,28 +37,90 @@ The <xref:Microsoft.AspNetCore.Components.Forms?displayProperty=fullName> namesp
 
+:::moniker range=">= aspnetcore-8.0"
+
+## Enhanced navigation and form handling
+
+Examples adopt [enhanced navigation](xref:blazor/fundamentals/routing#enhanced-navigation-and-form-handling) for form POST requests with the `Enhance` parameter for `EditForm` forms or the `data-enhance` attribute for HTML forms (`<form>`):
+
+```razor
+<EditForm Enhance ...>
+    ...
+</EditForm>
+```
+
+```html
+<form data-enhance ...>
+    ...
+</form>
+```
+
+<span aria-hidden="true">❌</span><span class="visually-hidden">Unsupported:</span> You can't set enhanced navigation on a form's ancestor element to enable enhanced navigation for the form.
+
+```html
+<div data-enhance>
+    <form ...>
+        <!-- NOT enhanced -->
+    </form>
+</div>
+```
+
+> [!WARNING]
+> ***Don't try to enhance form navigation for forms that POST to non-Blazor endpoints.***
+>
+> This is the primary reason why enabling form POSTs with enhanced navigation requires an explicit gesture on each `EditForm` and `<form>` element. It isn't safe to assume that all forms in general should use enhanced posts because any given form might not target a Blazor endpoint.
+
+To disable enhanced navigation for a form:
+
+* For an `EditForm`, remove the `Enhance` parameter from the form element (or set it to `false`: `Enhance="false"`).
+* For an HTML `<form>`, remove the `data-enhance` attribute from form element (or set it to `false`: `data-enhance="false"`).
+
+Blazor's enhanced navigation and form handing may undo dynamic changes to the DOM if the updated content isn't part of the server rendering. To preserve the content of an element, use the `data-permanent` attribute.
+
+In the following example, the content of the `<div>` element is updated dynamically by a script when the page loads:
+
+```html
+<div data-permanent>
+    ...
+</div>
+```
+
+To disable enhanced navigation and form handling globally, see <xref:blazor/fundamentals/startup#enhanced-navigation-and-form-handling>.
+
+Once Blazor has started on the client, you can use the `enhancedload` event to listen for enhanced page updates. This allows for re-applying changes to the DOM that may have been undone by an enhanced page update.
+
+```javascript
+Blazor.addEventListener('enhancedload', () => console.log('Enhanced update!'));
+```
+
+:::moniker-end
+
 ## Examples in this article
 
 :::moniker range=">= aspnetcore-8.0"
 
-Components are configured for interactivity with server rendering. For a client-side experience in a Blazor Web App, change the render mode in the `@attribute` directive at the top of the component to either:
+Components are configured for interactivity with server rendering and enhanced navigation. For a client-side experience in a Blazor Web App, change the render mode in the `@rendermode` directive at the top of the component to either:
```

Comment:
I think what we eventually want to do here is build up the user's understanding of form handling in Blazor by building off of concepts they probably already know, like the basics of forms in HTML. So, I think that means the initially sections should start with just a plan `<form>` (not `EditForm`) and then build up from there. First show how you can handle the form submission from the server using `@onsubmit`. Then introduce `EditForm` as a more convenient way to build a server based form. Then introduce enhanced form handling from the server. Then introduce interactive form handling. But we don't have to do this restructuring right away. It can wait until we get the .NET 8 stuff covered.

Reply:
Very well, but it's probably best to reorganize the coverage after RC2 releases, perhaps late next week. I'll open an issue to make sure the task doesn't get lost when this PR merges.